### PR TITLE
! data accessing nil

### PR DESCRIPTION
**Pull request by pape-io**
Patch version

3.7.0 introduced data accessing improvements but also improved the following different behaviour:

before 3.7.0
```
LHC.get('something').data.not_existing_key # nil
```

after 3.7.0
```
LHC.get('something').data.not_existing_key
# throws no method error
```

This one is especially nasty as it breaks rails auth: https://github.com/local-ch/rails-auth/blob/6c40ebdff156b240ca84aba57bc4a749a3a5cbf9/app/helpers/rails_auth/sessions_helper.rb#L194

### CODE REVIEW

File: lib/lhc/response/data.rb [hunk 1]
```diff
@@ -1,10 +1,11 @@
 # Response data is data provided through the response body
 # but made accssible in the ruby world
-class LHC::Response::Data
+class LHC::Response::Data < OpenStruct
 
   def initialize(response)
     @response = response
     set_dynamic_accessor_methods
+    @table = as_json
```
Comment (Averethel) [inline lib/lhc/response/data.rb]: Why is this needed? Is this related to the issue?

Comment (pape-io) [inline lib/lhc/response/data.rb]: @table is OpenStructs internal to the data... maybe could go with super -.- let me check

Comment (pape-io) [inline lib/lhc/response/data.rb]: used super! @Averethel
